core/src/main/scripts/importer/libImportOncokb.py

```diff
@@ -0,0 +1,236 @@
+                    end = int(match[0][2])
+                else:
+                    end = start
+            else:
+                match = re.findall(r'[A-Z]?([0-9]+)_[A-Z]?([0-9]+)(.+)', protein_change)
+                if len(match) > 0:
+                    start = int(match[0][0])
+                    end = int(match[0][1])
+                else:
+                    match = re.findall(r'([A-Z\*])([0-9]+)[A-Z]?fs.*', protein_change)
+                    if len(match) > 0:
+                        start = int(match[0][1])
+                        end = start
+                    else:
+                        match = re.findall(r'([A-Z]+)?([0-9]+)((ins)|(del)|(dup))', protein_change)
+                        if len(match) > 0:
+                            start = int(match[0][1])
+                            end = start
+    return {"start": start, "stop": end}
+
+
+def evaluate_driver_passenger(oncogenic):
+    """Translate the OncoKB 'oncogenic' value to Putative_Driver' or Putative_Passenger'"""
+    custom_annotation = 'Putative_Passenger'
+    if oncogenic.lower() in ['oncogenic', 'likely oncogenic', 'predicted oncogenic']:
+        custom_annotation = 'Putative_Driver'
+    return custom_annotation
+
+
+def get_first_line(file_handle):
+    """Read the first line of a file ignoring comment lines."""
+    while True:
+        first_line = file_handle.readline()
+        if not first_line.startswith('#'):
+            file_handle.close()
+            return first_line
+
+
+def find_meta_file_by_fields(study_dir, field_dict):
+    """In the study directory find the meta file that holds specified field values."""
+    filenames = os.listdir(study_dir)
+    meta_files = [ file for file in filenames if file.startswith('meta_')]
+    meta_files = [ study_dir + "/" + file for file in meta_files ]
+    for meta_file in meta_files:
+        fields = read_meta_file(meta_file)
+        match_found = True
+        for field_name, field_value in field_dict.items():
+            if field_name not in fields or fields[field_name] != field_value:
+                match_found = False
+        if match_found:
+            return os.path.join(study_dir, meta_file)
+    return None
+
+
+def find_data_file_from_meta_file(metafile_path):
+    """From specified meta file return the value of the 'data_filename' field."""
+    fields = read_meta_file(metafile_path)
+    return fields['data_filename']
+
+
+def read_meta_file(metafile_path):
+    """Read fields of specified meta into a dict."""
+    fields = {}
+    with open(metafile_path) as metafile:
+        for line in metafile:
+            if line == '\n':
+                continue
+            match = re.findall(r'\s*(\S+)\s*:\s*(\S+)', line)[0]
+            fields[match[0]] = match[1]
+    return fields
+
+def get_oncokb_cancer_genes():
+    """Do a call to OncoKB to retrieve the Cancer Gene List. """
+    request_url = "https://public.api.oncokb.org/api/v1/utils/allCuratedGenes"
```

Comment:
This is the api to curated gene list not cancer genes. Which one you like to use?

Reply:
Thanks for the catch. Initially this was cancer genes but we changed it to curated genes without modifying the names.